src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -59,27 +61,51 @@ public class ExprNamed extends PropertyExpression<Object, Object> {
 	@SuppressWarnings("null")
 	private Expression<String> name;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		setExpr(exprs[0]);
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[0].getReturnType().equals(ItemStack.class)) {
+			setExpr(exprs[0].getConvertedExpression(ItemType.class));
+		} else {
+			setExpr(exprs[0]);
+		}
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !isCreatable((InventoryType) object)) {
+				Skript.error(Utils.A(literal.toString()) + " cannot be created.");
+				return false;
+			}
+		}
 		return true;
 	}
 	
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		String name = this.name.getSingle(e);
+	protected Object[] get(final Event event, final Object[] source) {
+		String name = this.name.getSingle(event);
 		if (name == null)
-			return get(source, obj -> obj); // No name provided, do nothing
+			return get(source, object -> {
+				if (object instanceof InventoryType) {
+					InventoryType type = (InventoryType) object;
+					if (!isCreatable(type))
+						return null;
+					return Bukkit.createInventory(null, type);
+				}
+				return object; // Return the same ItemType they passed without applying a name.
```

Comment:
The other `get` below handles converting ItemStacks to ItemTypes. Do we need to handle that here too (to avoid an ArrayStoreException)? I'm not actually sure that ItemStacks could even be passed into this method though.

Reply:
I don't know, TheLimeGlass didn't do this in the original PR but there was another change made to this class since that added that converter *outside* the getter, so I merged the changes together but since I don't know what/why that's there I can't say whether it's needed.